encoder.go

```diff
@@ -36,8 +37,9 @@ func EncoderByName(str string) (Encoder, error) {
 	var ok bool
 	if len(str) == 0 {
 		return Encoder{-1}, fmt.Errorf("empty multibase encoding")
-	} else if len(str) == 1 {
-		base = Encoding(str[0])
+	} else if utf8.RuneCountInString(str) == 1 {
+		r, _ := utf8.DecodeRuneInString(str)
```

Comment:
Do you know what the perf impact of introducing proper UTF8 handling are at this point? Similar to the discussion in https://github.com/multiformats/js-multiformats/pull/187 it'd be a shame to make the common-path more expensive just to support a toy encoding. In JS at least, once we start needing more than the single byte range things get complicated and expensive pretty quickly (even though some UTF8 is included by default, it's just not consistent). I'm sure the perf impact will be tiny, but CID handling is a very common operation so we should be careful not to make it measurable.

Same comment for the decode path.

Reply:
First, unlike JS `string` are just copy on write byte slices making this path far less different than JS. (We don't have the slice and rejoin trick.)

Secondly, the cost here is `~ 2ns` for ASCII bases since modern branch predictors are crazy good and the UTF8 decode loop would break on the first iteration.

Here is a benchmark from the STD where it counts 10 ASCII characters in a string:
```
BenchmarkValidStringTenASCIIChars-12             	289691846	         4.055 ns/op
```
The performance for 1 char wont be `0.4ns` since that would be 1~2 clock cycles, for the multiple iterations things are pipelined away. But it wont be longer either.